test/specs/cli.js

```diff
@@ -77,6 +77,54 @@ describe('cli', function () {
         });
     });
 
+    it('should exit with a non-zero status code when lint warnings pass `--max-warnings`', function () {
```

Comment:
I'd like to see one more test where the max warnings is set to 2, and the warnings generated are 3. or some such. just for the sake of being thorough.